lib/puppet/provider/zone/solaris.rb

```diff
@@ -263,7 +263,8 @@ def start
 
       unless Puppet::FileSystem.exist?(sysidcfg)
         begin
-          File.open(sysidcfg, "w", 0600) do |f|
+          # For compatibility reasons use System encoding for this OS file
+          Puppet::FileSystem.open(sysidcfg, "w:#{Encoding.default_external.name}", 0600) do |f|
```

Comment:
It looks like the content of `sysidcfg` file is coming in as a zone resource property value, which will be UTF-8. Any risk to this conversion, you think?